includes/class-sensei-lesson.php

```diff
@@ -268,27 +268,50 @@ public function lesson_info_meta_box_content() {
 	}
 
 	/**
-	 * lesson_prerequisite_meta_box_content function.
+	 * Lesson_prerequisite_meta_box_content function.
 	 *
 	 * @access public
 	 * @return void
 	 */
 	public function lesson_prerequisite_meta_box_content() {
 		global $post;
-		// Get existing post meta
+		// Get existing post meta.
 		$select_lesson_prerequisite = get_post_meta( $post->ID, '_lesson_prerequisite', true );
-		// Get the Lesson Posts
-		$post_args   = array(
-			'post_type'        => 'lesson',
-			'posts_per_page'   => -1,
-			'orderby'          => 'title',
-			'order'            => 'ASC',
-			'exclude'          => $post->ID,
-			'suppress_filters' => 0,
-			'post_status'      => [ 'publish', 'draft' ],
-		);
+		// Get current Course.
+		$current_lesson_course = get_post_meta( $post->ID, '_lesson_course', true );
+
+		if ( $current_lesson_course ) {
+			// Get Lesson Posts from current Course.
+			$post_args = array(
+				'post_type'        => 'lesson',
+				'posts_per_page'   => -1,
+				'orderby'          => 'title',
+				'order'            => 'ASC',
+				'exclude'          => $post->ID,
+				'suppress_filters' => 0,
+				'post_status'      => [ 'publish', 'draft' ],
+				'meta_query'       => array(
+					array(
+						'key'     => '_lesson_course',
+						'value'   => $current_lesson_course,
+						'compare' => '=',
+					),
+				),
+			);
+		} else {
+			// Get the Lesson Posts.
+			$post_args = array(
```

Comment:
Instead of repeating the same values for `$post_args` twice, what about defining the common ones before the `if` statement, and then adding the new ones if `$current_lesson_course` has a value? Would save several lines of code.

Reply:
Done!